Source/Core/Core/PowerPC/JitArm64/JitAsm.cpp

```diff
@@ -97,32 +97,24 @@ void JitArm64::GenerateAsm()
 
   if (assembly_dispatcher)
   {
-    if (GetBlockCache()->GetFastBlockMap())
+    if (GetBlockCache()->GetEntryPoints())
     {
       // Check if there is a block
-      ARM64Reg pc_masked = ARM64Reg::X25;
-      ARM64Reg cache_base = ARM64Reg::X24;
+      ARM64Reg pc_and_msr = ARM64Reg::X25;
+      ARM64Reg cache_base = ARM64Reg::X27;
       ARM64Reg block = ARM64Reg::X30;
-      LSL(pc_masked, DISPATCHER_PC, 1);
-      MOVP2R(cache_base, GetBlockCache()->GetFastBlockMap());
-      LDR(block, cache_base, pc_masked);
+      LDR(IndexType::Unsigned, pc_and_msr, PPC_REG, PPCSTATE_OFF(msr));
+      AND(pc_and_msr, pc_and_msr, LogicalImm(JitBaseBlockCache::JIT_CACHE_MSR_MASK, 32));
+      // The entry points map is indexed by ((msrBits << 26) | (address >> 2) << 3)
+      // The 3 corresponds to an 8 byte 64-bit pointer and that means we need to shift
+      // msr left by 29 bits and address left by 1 bit to get the index to the map.
+      LSL(pc_and_msr, pc_and_msr, 29);
+      ORR(pc_and_msr, pc_and_msr, DISPATCHER_PC, ArithOption(DISPATCHER_PC, ShiftType::LSL, 1));
+      MOVP2R(cache_base, GetBlockCache()->GetEntryPoints());
+      LDR(block, cache_base, pc_and_msr);
```

Comment:
Optimization (untested):

```suggestion
      // The entry points map is indexed by ((msrBits << 26) | (address >> 2) << 3)
      // The 3 corresponds to an 8 byte 64-bit pointer
      LDR(IndexType::Unsigned, pc_and_msr, PPC_REG, PPCSTATE_OFF(msr));
      MOVP2R(cache_base, GetBlockCache()->GetEntryPoints());
      UBFIZ(pc_and_msr, pc_and_msr, 26, 6);
      BFXIL(pc_and_msr, DISPATCHER_PC, 2, 30);
      LDR(block, cache_base, ArithOption(pc_and_msr, true));
```

Reply:
It works! :)